airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -399,4 +419,164 @@ private static long getEpoch(Record record, String fieldName) {
+    }
+    return result;
+  }
+
+  /**
+   * List tables from @param schema and @return their names
+   */
+  private List<String> listTables(final String schema) throws IOException {
+    if (schema != null) {
+      return database.query(context -> context.meta().getSchemas(schema).stream()
+          .flatMap(s -> context.meta(s).getTables().stream())
+          .map(Named::getName)
+          .collect(Collectors.toList()));
+    } else {
+      return List.of();
+    }
+  }
+
+  private Stream<JsonNode> exportTable(final String schema, final String tableName) throws IOException {
+    final String tableSql = String.format("%s.%s", schema, tableName);
+    try (final Stream<Record> records = database.query(ctx -> ctx.select(DSL.asterisk()).from(tableSql).fetchStream())) {
+      return records.map(record -> {
+        final Set<String> jsonFieldNames = Arrays.stream(record.fields())
+            .filter(f -> f.getDataType().getTypeName().equals("jsonb"))
+            .map(Field::getName)
+            .collect(Collectors.toSet());
+        final JsonNode row = Jsons.deserialize(record.formatJSON(DB_JSON_FORMAT));
+        // for json fields, deserialize them so they are treated as objects instead of strings. this is to
+        // get around that formatJson doesn't handle deserializing them for us.
+        jsonFieldNames.forEach(jsonFieldName -> ((ObjectNode) row).replace(jsonFieldName, Jsons.deserialize(row.get(jsonFieldName).asText())));
+        return row;
+      });
+    }
+  }
+
+  @Override
+  public void importDatabase(final Map<String, Stream<JsonNode>> data) throws IOException {
+    importDatabase(DEFAULT_SCHEMA, data, true);
+  }
+
+  private void importDatabase(final String targetSchema, final Map<String, Stream<JsonNode>> data, boolean useIntermediateSchema) throws IOException {
+    if (!data.isEmpty()) {
+      if (useIntermediateSchema) {
+        final String tempSchema = "import_staging_" + RandomStringUtils.randomAlphanumeric(5);
+        try {
+          dropSchema(tempSchema);
+          createSchema(tempSchema);
+          initSchema(tempSchema);
+          for (final String tableName : data.keySet()) {
+            importTable(tempSchema, tableName, data.get(tableName));
+          }
+          swapSchema(tempSchema, targetSchema);
+        } finally {
+          dropSchema(tempSchema);
+        }
+      } else {
+        for (final String tableName : data.keySet()) {
+          importTable(DEFAULT_SCHEMA, tableName, data.get(tableName));
+        }
+      }
+      // TODO write "import success vXX on now()" to audit log table?
+    } else {
+      throw new IllegalArgumentException("Import empty Airbyte database");
+    }
+  }
+
+  private void createSchema(final String schema) throws IOException {
+    database.query(ctx -> ctx.execute(String.format("CREATE SCHEMA IF NOT EXISTS %s;\n", schema)));
+  }
+
+  private void initSchema(final String schema) throws IOException {
```

Comment:
i think this is brittle. if someone doesn't realize this is here and changes the capitalization in the comment then this will break when someone tries to import there data. i think the schema swap stuff is adding uncertainty and very little value. 

we should truncate the tables we are going to replace. it is fine to name them explicitly in this class as constant. e.g. `PERSISTENCE_TABLES = Sets.newHashset("Jobs", "Attempts", "AirbyteMetdata");` and we just truncate all of these tables.